hs_core/views/__init__.py

```diff
@@ -527,6 +527,7 @@ def set_resource_flag(request, shortkey, *args, **kwargs):
     elif t == 'make_shareable':
        _set_resource_sharing_status(request, user, res, flag_to_set='shareable', flag_value=True)
 
+    request.session['resource-mode'] = request.POST.get('resource-mode', 'view')
```

Comment:
@pkdash In other words, wouldn't be sufficient just to directly set ```resource-mode``` session variable to ```edit``` here? Specifically I don't see where ```resource-mode``` is set to ```edit``` in the new added code. All I see is the hidden input field on the templates to hold the ```resource-mode``` session variables. Perhaps it is set in JavaScript functions? I don't see them from the code changes though.